WordPress/Classes/ViewRelated/Comments/Content/CommentContentHeaderView.swift

```diff
@@ -0,0 +1,197 @@
+import SwiftUI
+import DesignSystem
+
+struct CommentContentHeaderView: View {
+
+    let config: Configuration
+
+    var body: some View {
+        HStack(spacing: .DS.Padding.split) {
+            AvatarsView(style: .single(config.avatarURL))
+            VStack(alignment: .leading) {
+                Text(config.username)
+                    .style(.bodySmall(.regular))
+                    .foregroundStyle(Color.DS.Foreground.primary)
+                    .lineLimit(1)
+                Text(config.handleAndTimestamp)
+                    .style(.bodySmall(.regular))
+                    .foregroundStyle(Color.DS.Foreground.secondary)
+                    .lineLimit(1)
+            }
+            Spacer()
+            CommentContentHeaderMenu(menu: config.menu)
+        }
+    }
+
+    struct Configuration {
+        let avatarURL: URL?
+        let username: String
+        let handleAndTimestamp: String
+        let menu: [[MenuItem]]
+    }
+
+    typealias MenuItem = CommentContentHeaderMenu.Item
+}
+
+extension CommentContentHeaderView.Configuration {
+
+     init(avatarURL: URL?, username: String, handleAndTimestamp: String, menu: [CommentContentHeaderView.MenuItem]) {
+        self.init(
+            avatarURL: avatarURL,
+            username: username,
+            handleAndTimestamp: handleAndTimestamp,
+            menu: [menu]
+        )
+    }
+}
+
+struct CommentContentHeaderMenu: View {
+   let menu: [[Item]]
+
+   var body: some View {
+       Menu {
+           ForEach(Array(menu.enumerated()), id: \.offset) { sectionIndex, section in
+               if !section.isEmpty {
+                   Section {
+                       ForEach(Array(section.enumerated()), id: \.offset) { itemIndex, menuItem in
+                           button(for: menuItem)
+                       }
+                   }
+                   if sectionIndex < menu.count - 1 && !menu[sectionIndex + 1].isEmpty {
```

Comment:
Could we improve this by using a compactMap or filter logic to remove sections which have no items? Then we could always place the divider after the last index available.